File: LeetcodeEasyProblems/_169. Majority Element/Solution.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        candidate = 0
        count = 0

        for i in nums:
            if count == 0:
                candidate = i
            if candidate == i:
                count+= 1
            else:
                count-=1
            
        return candidate

File: LeetcodeEasyProblems/_169. Majority Element/Solution.java
public class Solution {
    public int majorityElement(int[] nums) {
        int candidate = 0;
        int count = 0;
        for (int element : nums){
            if (count == 0){
                candidate = element;
            }
            if(candidate == element){
                count++;
            }else{
                count--;
            }

        }

        return candidate;
    }
}
